yaml-streamly/src/Data/Yaml/Internal.hs

```diff
@@ -215,55 +208,50 @@ data ParseState = ParseState {
 
 type Parse = StateT ParseState IO
 
-requireEvent :: Event -> Parser (ReaderT JSONPath Parse) Event ()
+requireEvent :: Event -> ParserK Event Parse ()
 requireEvent e = do
-    f <- toParserK anyEvent
+    f <- ParserK.adapt anyEvent
     unless (f == Just e) $ missed (Just e)
 
 {-# INLINE anyEvent #-}
-anyEvent :: MonadCatch m => ParserD.Parser m a (Maybe a)
-anyEvent = ParserD.Parser step initial extract
-  where
-  initial = pure $ ParserD.IPartial ()
-  step _ a = pure $ ParserD.Done 0 (Just a)
-  extract _ = pure Nothing
-
+anyEvent :: MonadCatch m => Parser.Parser a m (Maybe a)
+anyEvent = Parser.fromFold Fold.one
```

Comment:
Make it ParserK using ParserK.adapt. That will avoid using adapt wherever it is used.